### PR TITLE
Move createCopy method from UniquePersonList to respective classes

### LINKED ISSUE #68: Move copyOrder() from UniquePersonList to OrderList

### PR DESCRIPTION
**Pull request by awdse22**
Close #68 

* Moved `copyPerson` method from `UniquePersonList` to `Person`
* Added `createCopy` method in `OrderList`
* Moved `copyOrder` method from `UniquePersonList` to `Order`

### CODE REVIEW

File: src/main/java/seedu/sellsavvy/model/person/Person.java [hunk 1]
```diff
@@ -156,4 +156,17 @@ public String toString() {
...
+     * Creates a new copy of this {@code Person} with new attributes.
+     * Only the tags are not copied.
+     */
+    public Person createCopy() {
+        Name name = new Name(getName().toString());
+        Address address = new Address(getAddress().toString());
+        Phone phone = new Phone(getPhone().toString());
+        Email email = new Email(getEmail().toString());
+        Set<Tag> tags = getTags();
+        OrderList orderList = getOrderList().createCopy();
```
Comment (FooChao) [inline src/main/java/seedu/sellsavvy/model/person/Person.java:169]: Can u make the predicate same as well?

**Review by FooChao — COMMENTED** [review]
I feel that we should copy the predicate as well and maybe check if predicate same as well for equals (in future or u can do it as well if u want) for Person class

File: src/main/java/seedu/sellsavvy/model/person/Person.java (hunk 1 above)
Comment (awdse22) [reply src/main/java/seedu/sellsavvy/model/person/Person.java:169]: there is no predicate field for Person but i'll copy over from filteredOrders and apply it to the new created Person object

**Review by FooChao — COMMENTED** [review]
Just in case u didn't notice a lot of test cases failed

File: src/main/java/seedu/sellsavvy/model/person/Person.java [hunk 2]
```diff
@@ -156,4 +156,26 @@ public String toString() {
...
+        Person personCopy = new Person(name, phone, email, address, tags, orderList);
+
+        if (filteredOrders.getPredicate() == null) {
+            return personCopy;
+        }
+
+        @SuppressWarnings("unchecked")
+        Predicate<Order> filterPredicate = (Predicate<Order>) filteredOrders.getPredicate();
```
Comment (kaixuan477) [inline src/main/java/seedu/sellsavvy/model/person/Person.java:177]: I used assigned to  Predicate<? super Order> for predicate to not use @suppresswarnings, is that better?

File: src/main/java/seedu/sellsavvy/model/order/OrderList.java [hunk 3]
```diff
@@ -144,4 +144,15 @@ private int indexOf(Order target) {
 
         return -1;
     }
+
+    /**
+     * Creates a copy of this {@code OrderList} with new Orders.
+     */
+    public OrderList createCopy() {
+        OrderList orderList = new OrderList();
```
Comment (kaixuan477) [inline src/main/java/seedu/sellsavvy/model/order/OrderList.java:152]: Rename orderList to orderListCopy for consistency?

**Review by kaixuan477 — APPROVED** [review]
Looks good, just small edits.

File: src/main/java/seedu/sellsavvy/model/person/Person.java (hunk 2 above)
Comment (awdse22) [reply src/main/java/seedu/sellsavvy/model/person/Person.java:177]: image
should i just change the method signature for `updateFilteredOrderList`?

File: src/main/java/seedu/sellsavvy/model/order/OrderList.java (hunk 3 above)
Comment (awdse22) [reply src/main/java/seedu/sellsavvy/model/order/OrderList.java:152]: sure

File: src/main/java/seedu/sellsavvy/model/person/Person.java (hunk 2 above)
Comment (awdse22) [reply src/main/java/seedu/sellsavvy/model/person/Person.java]: changed

**Pull request merged**